modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -272,11 +273,23 @@ INSFVRhieChowInterpolator::initialSetup()
         .template condition<AttribSysNum>(_u->sys().number())
         .queryInto(var_objects);
     for (auto * const var_object : var_objects)
-      if (!dynamic_cast<INSFVMomentumResidualObject *>(var_object))
+    {
+      // Allow FVElementalKernel that are not INSFVMomentumResidualObject for now, refs #20695
+      if (!dynamic_cast<INSFVMomentumResidualObject *>(var_object) &&
+          !dynamic_cast<FVElementalKernel *>(var_object))
         mooseError("Object ",
                    var_object->name(),
                    " is not a INSFVMomentumResidualObject. Make sure that all the objects applied "
                    "to the momentum equation are INSFV or derived objects.");
+      else if (!dynamic_cast<INSFVMomentumResidualObject *>(var_object) &&
+               dynamic_cast<FVElementalKernel *>(var_object))
+        mooseWarning(
+            "Elemental kernel ",
+            var_object->name(),
+            " is not a INSFVMomentumResidualObject. Make sure that all the objects applied "
+            "to the momentum equation are INSFV or derived objects.");
```

Comment:
If we are adding objects into the module that are meant to be used in INSFV simulations and a user users them, then I don't think we should even warn about it. I think we should have it one way or the other: allow non-INSFVMomentumResidual objects in the momentum equation and don't warn about it; or don't allow INSFVMomentumResidualObjects in the momentum equation.

Reply:
If we don't allow non-INSFVMomentumResidualObjects into the module, then I am good with keeping the warning ... that means a user has written their own object and they've been lazy and decided not to implement `gatherRCData`. Such laziness is 100% justified if the object's residual does not depend on velocity, since it doesn't produce anything that should go into 'a' ... but it would be unjustified if their residual does indeed depend on 'a'. Not knowing whether it's justified or not *a priori* (we have no idea what the user's custom residual looks like) I think the warning makes total sense.

But we can control the residuals in the module, and so we shouldn't warn the user if they use our own stuff.

Reply:
So I suppose since I've asked that you add `gatherRCData` to the new object you've added here, then we're still all-clear in the module on having all momentum objects implement `gatherRCData` ... in which case I would suggest that we keep the warning ... but actually remove the error. Because a user could also add a `FVFluxBC` to the momentum equation that doesn't depend on the velocity dofs and that would be ok

Reply:
So TLDR ... this is fine 😆